external-crates/move/crates/move-compiler/tests/custom_rules/redundant_conditional.move

```diff
@@ -0,0 +1,17 @@
+module 0x42::M {
+
+    fun func1() {
+        let x = true;
+        if (x) {
+            false
+        } else {
+            true
+        };
```

Comment:
Should add cases for things like `if (foo()) true else false` and `if (foo()) (true) else (false)`